libbeat/autodiscover/providers/kubernetes/service.go

```diff
@@ -74,8 +74,9 @@ func NewServiceEventer(uuid uuid.UUID, cfg *conf.C, client k8s.Interface, publis
 
 	if metaConf.Namespace.Enabled() || config.Hints.Enabled() {
 		namespaceWatcher, err = kubernetes.NewNamedWatcher("namespace", client, &kubernetes.Namespace{}, kubernetes.WatchOptions{
-			SyncTimeout: config.SyncPeriod,
-			Namespace:   config.Namespace,
+			SyncTimeout:  config.SyncPeriod,
+			Namespace:    config.Namespace,
+			HonorReSyncs: true,
```

Comment:
Wondering if we need to make this configurable? 
I dont find a place where we have documented the use of resyncs

Reply:
I think it is just for devs, indeed.